salt/dex/init.sls

```diff
@@ -85,12 +85,16 @@ include:
       - file: /root/dex.yaml
 
 dex_secrets:
-  cmd.run:
+  caasp_cmd.run:
     - name: |
-        until kubectl get secret dex-tls --namespace=kube-system ; do
-            kubectl create secret generic dex-tls --namespace=kube-system --from-file=/etc/pki/dex.crt --from-file=/etc/pki/dex.key
-            sleep 5
-        done
+        kubectl create secret generic dex-tls --namespace=kube-system --from-file=/etc/pki/dex.crt --from-file=/etc/pki/dex.key
+    - unless: |
+        kubectl get secret dex-tls --namespace=kube-system
+    - retry:
+        until: |
+          kubectl get secret dex-tls --namespace=kube-system
```

Comment:
If `kubectl create` returns a 0 errcode it's supposed to have created this secret, so I think we can remove the `until` here. The `retry` will only retry if the errcode is not 0.

Reply:
We are keeping the same logic we previously had: we cannot trust `kubectl create/apply` because of concurrency issues. So we must repeat the command (regardless of the output) until the `kubectl get` succeeds.